src/database/database-sqlite3.h

```diff
@@ -64,10 +64,10 @@ class Database_SQLite3 : public Database
 		sqlite3_vrfy(sqlite3_bind_double(s, iCol, val));
 	}
 
-	inline std::string sqlite_to_string(sqlite3_stmt *s, int iCol)
+	inline std::string_view sqlite_to_string(sqlite3_stmt *s, int iCol)
 	{
 		const char* text = reinterpret_cast<const char*>(sqlite3_column_text(s, iCol));
-		return std::string(text ? text : "");
+		return text ? std::string_view(text) : std::string_view();
```

Comment:
I think this change risks introducing regressions if this isn't used very carefully. From the [sqlite3 docs](https://www.sqlite.org/c3ref/column_blob.html):

> The pointers returned are valid until a type conversion occurs as described above, or until [sqlite3_step()](https://www.sqlite.org/c3ref/step.html) or [sqlite3_reset()](https://www.sqlite.org/c3ref/reset.html) or [sqlite3_finalize()](https://www.sqlite.org/c3ref/finalize.html) is called. The memory space used to hold strings and BLOBs is freed automatically.

Hence, if we want to maintain ownership of the value, we can't return a string view here, we want to copy and return a string owned by us (rather than by SQLite).

We could still do zero-copy here, but then this should IMO be a new method `sqlite_to_string_view` with a big fat warning, as otherwise this is a footgun (especially compared to the previous code).

Reply:
We always follow the step -> read values -> reset pattern and nothing else would make sense anyway as you can only retrieve the right values from the stmt before calling another method.
It will also be obvious to any attentive programmer that a string_view must either be handed off to somewhere else *or* cloned right in the moment you have it and cannot be stored (not any more dangerous than handling `sqlite3_column_blob`).

I can rename the function if you want but I don't see the big risk.

Reply:
Fair. I just checked the usages to be sure and they all seem proper.

I think a comment regarding the validity of the string view might be good to have though. A rename `string` -> `stringview` would also help readability. In many contexts the type is hidden, and `sqlite_to_string` could reasonably be assumed to return a string (which would then be owned by you) rather than a string view (which implies you would have to pay attention to its lifetime).